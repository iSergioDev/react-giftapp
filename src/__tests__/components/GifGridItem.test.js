import { shallow } from 'enzyme'
import React from 'react'
import GifGridItem from '../../components/GifGridItem'

describe('Pruebas en <GifGridItem />', () => {
    const title = 'Título Test'
    const url = 'https://localhost/test.jgp'
    const wrapper = shallow( <GifGridItem title={title} url={url} />)

    test('Debe mostrar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot()
    })

    test('Debe de tener un párrafo con el title', () => {

        const p = wrapper.find('p')
        expect(p.text().trim()).toBe( title )
    })

    test('Debe de tener la iamgen igual a la url y alt de los props', () => {
        const img = wrapper.find('img') 
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)

    })

    test('Debe de tener la clase CSS animate_fadeIn', () => {
        const div  = wrapper.find('div')
        const className = div.prop('className')
        
        expect(className.includes('animate__fadeIn')).toBe(true)

    })
})
