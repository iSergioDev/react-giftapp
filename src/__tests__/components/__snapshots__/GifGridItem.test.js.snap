// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Título Test"
    src="https://localhost/test.jgp"
  />
  <p>
    Título Test
  </p>
</div>
`;
